synapse/federation/transport/server.py

```diff
@@ -542,6 +543,22 @@ async def on_PUT(self, origin, content, query, room_id, event_id):
         return 200, content
 
 
+class FederationMakeKnockServlet(BaseFederationServlet):
+    PATH = "/make_knock/(?P<room_id>[^/]*)/(?P<user_id>[^/]*)"
+
+    async def on_GET(self, origin, content, query, room_id, user_id):
+        content = await self.handler.on_make_knock_request(origin, room_id, user_id)
+        return 200, content
+
+
+class FederationV1MakeKnockServlet(BaseFederationServlet):
```

Comment:
I think this is now v2?

```suggestion
class FederationV2MakeKnockServlet(BaseFederationServlet):
```